tests/bluechi_test/machine.py

```diff
@@ -142,16 +151,17 @@ def copy_machine_lib(self):
                     target_dir = os.path.join("/", "tmp", "bluechi_machine_lib")
                     self.create_file(target_dir, filename, content)
 
-    def restart_with_config_file(self, config_file_location, service):
-        unit_dir = "/usr/lib/systemd/system"
-        service_file = f"{service}.service"
+    def restart_with_config_file(self, cfg_file_location: str, svc_name: str,
+                                 svc_dir: str = "/usr/lib/systemd/system") -> None:
 
-        self._track_changed_file(unit_dir, service_file)
-        self.client.exec_run(f"sed -i '/ExecStart=/c\\ExecStart=/usr/libexec/{service} -c "
-                             f"{config_file_location}' "
-                             f"{os.path.join(unit_dir, service_file)}")
-        self.systemctl.daemon_reload()
-        self.systemctl.restart_unit(service_file)
+        service = Service(svc_name, svc_dir)
+
+        self.load_systemd_service(service)
+        service.set_option(Section.Service, Option.ExecStart,
+                           service.get_option(Section.Service, Option.ExecStart) + f" -c {cfg_file_location}")
+
+        self.install_systemd_service(service, backup_existing=True, stop_at_cleanup=False)
+        self.systemctl.restart_unit(svc_name)
```

Comment:
This function only makes sense for `bluechi-agent` and `bluechi-controller`. So I think it is implemented too general. What do you think about having:
```python
class BluechiService(Service):
# ...
class BluechiAgentService(BluechiService):
# ...
class BluechiControllerService(BluechiService):
#...

# and changing function to:
def restart_with_config_file(self, cfg_file_location: str, bluechi_service: BluechiService) -> None:
```

Reply:
I don't see a point in creating subclasses, which don't add any logic, just predefined service location and name.

Reply:
The point in this case is to limit the usage of `restart_with_config_file` to `bluechi-agent` and `bluechi-controller` - it makes no sense (and rather introduces bugs) to use any other service here and add `-c <path>` to its ExecStart. The `restart_with_config_file` function could also be part of the `BluechiService` class (and the machine will be passed as parameter) so it wouldn't be empty - and thinking about the other tests for the parameters (`-p` etc.) it would make sense to provide a function for each one on the bluechi class and encapsulate common functionality (like install + restart).

Reply:
OK, I will try to take a look how to make restart_with_config_file usable for all usage (pretty much focus on updating the service file and restart a service, while service file changes would be done within each test)

Reply:
I really like that the `restart_with_config_file` is now a simple load + install systemd unit